wp_api/src/login/login_client.rs

```diff
@@ -162,35 +174,41 @@ impl WpLoginClient {
         }
     }
 
-    async fn attempt_is_wordpress_site(
-        &self,
-        attempt_site_url: &str,
-        api_link_header_result: Result<FindApiRootLinkHeaderSuccess, FindApiRootLinkHeaderFailure>,
-        parse_html_result: Result<IsWordPressSiteParseHtmlResult, ParseHtmlFailure>,
-    ) -> IsWordPressSiteAttemptResult {
-        let fetch_wp_json_result = self.fetch_wp_json(attempt_site_url).await;
-        IsWordPressSiteAttemptResult {
-            api_link_header_result,
-            fetch_wp_json_result,
-            parse_html_result,
-        }
-    }
-
     async fn find_api_root_url(
         &self,
-        attempt_site_url: &str,
+        parsed_site_url: ParsedUrl,
         api_root_url_from_link_tag: Option<ParsedUrl>,
+        fetch_wp_json_result: Result<FetchWpJsonSuccess, FetchWpJsonFailure>,
     ) -> Result<FindApiRootLinkHeaderSuccess, FindApiRootLinkHeaderFailure> {
-        let parsed_site_url = ParsedUrl::parse(attempt_site_url)
-            .map_err(|error| FindApiRootLinkHeaderFailure::ParseSiteUrl { error })?;
         if let Some(api_root_url) = api_root_url_from_link_tag {
-            Ok(FindApiRootLinkHeaderSuccess {
+            return Ok(FindApiRootLinkHeaderSuccess {
                 parsed_site_url,
                 api_root_url,
-            })
-        } else {
-            self.fetch_and_parse_api_root_url_header(parsed_site_url)
-                .await
+            });
+        }
+        match self
+            .fetch_and_parse_api_root_url_header(parsed_site_url.clone())
+            .await
+        {
+            Ok(s) => Ok(s),
+            Err(fetch_and_parse_api_root_url_header_err) => {
+                // If we can't find the link header, but we were able to fetch `/wp-json` from the
+                // attempt url, we assume that it's the correct api root url
+                //
+                // We don't immediately rely on this because there might be cases where there are
+                // multiple `/wp-json` paths in which case we want to prioritize the one returned
+                // from the link header
```

Comment:
More specifically, there might be a `/wp-json` path _and_ the actual REST API root (which could be something else).

Not sure if we _need_ to update this comment – I'll leave that to you

Reply:
Addressed in 30cfa0427745e558e20a4d9cca693aa45e468e6c.